app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentPresenter.kt

```diff
@@ -91,22 +96,67 @@ class FeatureFlagsFragmentPresenter @Inject constructor(
   }
 
   private fun onBackNavigation() {
-    val overriddenFeatureFlags = featureFlagStates.map { (id, value) ->
-      OverriddenFeatureFlag.newBuilder()
-        .setId(id)
-        .setOverriddenValue(value)
-        .build()
+    val overriddenFlags = computeOverriddenFlags()
+    val resetFlags = featureFlagsViewModel.resetFlags.value?.keys?.toList().orEmpty()
+
+    when {
+      resetFlags.isNotEmpty() -> applyResetsThenOverrides(overriddenFlags)
+      overriddenFlags.isNotEmpty() -> overrideFeatureFlags(overriddenFlags)
+      else -> activity.finish()
     }
+  }
+
+  private fun computeOverriddenFlags(): List<OverriddenFeatureFlag> {
+    return featureFlagsViewModel.featureFlagStates.value
+      ?.filter { (id, value) ->
+        featureFlagsViewModel.resetFlags.value?.get(id) != value
+      }
+      ?.map { (id, value) ->
+        OverriddenFeatureFlag.newBuilder()
+          .setId(id)
+          .setOverriddenValue(value)
+          .build()
+      }
+      .orEmpty()
+  }
```

Comment:
Nit: Can be fit in a single line?
```suggestion
  private fun computeOverriddenFlags(): List<OverriddenFeatureFlag> {
    return featureFlagsViewModel.featureFlagStates.value
      ?.filter { (id, value) -> featureFlagsViewModel.resetFlags.value?.get(id) != value }
      ?.map { (id, value) ->
        OverriddenFeatureFlag.newBuilder()
          .setId(id)
          .setOverriddenValue(value)
          .build()
      }.orEmpty()
  }
```

Reply:
updated